src/main/java/uk/gov/fco/documentupload/service/ocr/OCRService.java

```diff
@@ -0,0 +1,134 @@
+        }
+        return true;
+    }
+
+    public String extractData(FileUpload upload) throws IOException {
+        if (enabled) {
+            log.info("Starting data extraction");
+            try {
+                SdkBytes bytes = SdkBytes.fromInputStream(upload.getInputStream());
+                Document doc = Document.builder().bytes(bytes).build();
+
+                List<FeatureType> featureTypes = new ArrayList<>();
+                featureTypes.add(FeatureType.QUERIES);
+
+                Query firstNameQuery = Query.builder().alias("firstName").text("Retrieve the subject's first name").build();
+                Query lastNameQuery = Query.builder().alias("lastName").text("Retrieve the subject's surname").build();
+                Query dobQuery = Query.builder().alias("dob").text("Retrieve the user's date of birth").build();
+
+                Collection<Query> queries = new ArrayList<>(Arrays.asList(firstNameQuery, lastNameQuery, dobQuery));
+
+                QueriesConfig queryConfig = QueriesConfig.builder().queries(queries).build();
+
+                AnalyzeDocumentRequest request = AnalyzeDocumentRequest.builder().featureTypes(featureTypes).document(doc).queriesConfig(queryConfig).build();
+
+                AnalyzeDocumentResponse response = textract.analyzeDocument(request);
+
+                List<Block> blocks = response.blocks();
+                List<Block> queryBlocks = this.getBlocksByType(blocks, BlockType.QUERY);
+                List<Block> queryResultBlocks = this.getBlocksByType(blocks, BlockType.QUERY_RESULT);
+
+                HashMap<String, String> results = mapQueriesToAnswers(queryBlocks, queryResultBlocks);
+
+                return new ObjectMapper().writeValueAsString(results);
+
+            } catch (TextractException e) {
+                log.error(e.getMessage());
+            }
+        }
+        return "{}";
+    }
+
+    private List<Block> getBlocksByType(List<Block> blocks, BlockType type) {
+        List<Block> queryBlocks = new ArrayList<>();
+        for (Block block : blocks) {
+            if (block.blockType() == type) {
+                queryBlocks.add(block);
+            }
+        }
+        return queryBlocks;
+    }
+
+    private String getAnswerId(List<Relationship> relationships) {
+        String id = "";
+        for (Relationship relationship : relationships) {
+            if (relationship.type() == RelationshipType.ANSWER) {
+                id = relationship.ids().get(0);
+                break;
+            }
+        }
+        return id;
+    }
+
+    private String getAnswer(List<Block> blocks, String blockId) {
+        String answer = "";
+        for (Block block : blocks) {
+            if (Objects.equals(block.id(), blockId)) {
+                answer = block.text();
+                break;
+            }
+        }
+        return answer;
+    }
+
+    private HashMap<String, String> mapQueriesToAnswers(List<Block> queryBlocks, List<Block> queryResultBlocks) {
```

Comment:
Mentioned that we're shelving this for now, but in future

let's change to something like 

```suggestion
    private HashMap<String, String> mapQueriesToAnswersFromResponse(response AnalyzeDocumentResponse) {
        List<Block> queryBlocks = blocks.stream().filter(block -> block.blockType() == BlockType.QUERY).collect(Collectors.toList());
       // then iterate over queryBlocks to get relationship and answer
    }
```
I think you can use something like .findFirst(..) rather than get(0) as well